Lib/test/test_grammar.py

```diff
@@ -70,8 +70,8 @@ def test_plain_integers(self):
                 warnings.filterwarnings('error', category=Py3xWarning)
                 with self.assertRaises(Py3xWarning) as oct:
                     compile('032', '<test string>', 'eval')
-                self.assertIn("octal literals are not supported in 3.x;\n" 
-                              "drop the leading 0",
+                self.assertIn("ctal literals are not supported in 3.x:\n" 
+                              "use an 0o prefix for octal integers or drop the leading '0'",
```

Comment:
I don't think either of these are fixes?

Reply:
I wanted to actually comment on this, I find in Python 2.6+ `01` resolves to `1` so removing the leading zero may be a fix, no? In 3.x we use `0o`, I stand to be corrected though :)

Reply:
Ah, so octal literals *are* supported in python 3? If so, I'm a bit confused. (`0o123` evaluates to 83 in both Python 2.7.18 and 3.9.10/.)

Reply:
> Ah, so octal literals _are_ supported in python 3? If so, I'm a bit confused. (`0o123` evaluates to 83 in both Python 2.7.18 and 3.9.10/.)

Yes, for Py 2.6+, this is correct.

I think support for syntax like `01` was left for compatibility in 2.6+ but resolves to `1`.

Reply:
My understanding is the `0o*` syntax was meant to replace `0*` in 2.6+, do you think so too?

Reply:
This turns out be an interesting mess!

python 2.7:

```
$ python2.7
Python 2.7.18 (default, Jan 24 2022, 13:30:27) 
[GCC OpenBSD Clang 13.0.0] on openbsd7
Type "help", "copyright", "credits" or "license" for more information.
>>> 0123
83
>>> 0o123
83
>>>
```

python 3.9:

```
$ python3
Python 3.9.10 (main, Jan 24 2022, 13:28:02) 
[Clang 13.0.0 ] on openbsd7
Type "help", "copyright", "credits" or "license" for more information.
>>> 0123
  File "<stdin>", line 1
    0123
       ^
SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers
>>> 0o123
83
>>>
```

So I think the message is misleading: octal literals *are* supported in Python 3 but of the two ways you could express them in Python 2.7 (`0` and `0o`) only `0o` is still supported.

Reply:
Thats true, I see, foolish me was just testing `01`, `02`.  Will do the needful.

Reply:
So, if I follow:

> use an 0o prefix for octal integers or drop the leading '0'

is still correct(?), but I'd probably suffix "if you intended the integer to be decimal".

Reply:
I have added this suffix now

Reply:
Thanks -- this looks good now. [Normally it helps to say "fixed in commit XYZ" to help the reviewer, but in this case I found the commit easily!]